src/com/civpvp/attrhider/AttrHider.java

```diff
@@ -33,129 +26,84 @@
+				}
+			});
+		
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EFFECT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					if(event.getPacket().getIntegers().read(0) != event.getPlayer().getEntityId()) {
+						event.getPacket().getBytes().write(0, (byte) 26);
+						event.getPacket().getBytes().write(1, (byte) 420);
+						event.getPacket().getBytes().write(2, (byte) 0);
+						event.getPacket().getIntegers().write(1, 420);
+					}
+				}
+			});
+		
 		ProtocolLibrary.getProtocolManager().addPacketListener(
-		new PacketAdapter(this, ListenerPriority.NORMAL, new PacketType[] { PacketType.Play.Server.ENTITY_METADATA }) {
-			public void onPacketSending(PacketEvent event) {
-				try {
-					Player observer = event.getPlayer();
-					//Get the entity from the packet
-					Entity entity = (Entity)event.getPacket().getEntityModifier(observer.getWorld()).read(0);
-					//If the entity is not the observer, and the entity is alive, and the entity is not a dragon or wither,
-					//and the entity is not the observer's mount
-					if ((entity != null) && (observer != entity) && ((entity instanceof LivingEntity)) &&
-					  (!(entity instanceof EnderDragon) && !(entity instanceof Wither)) && (entity.getPassenger() != observer)) {
-						event.setPacket(event.getPacket().deepClone());
-						StructureModifier<List<WrappedWatchableObject>> watcher = event.getPacket().getWatchableCollectionModifier();
-					  	for (WrappedWatchableObject watch : watcher.read(0)) {
-							if ((watch.getIndex() == 6) && (((Float)watch.getValue()).floatValue() > 0.0F)) {
-								watch.setValue(20f);
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_METADATA) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					Player player = event.getPlayer();
+					Entity entity = getEntityById(event.getPlayer().getWorld(), event.getPacket().getIntegers().read(0));
+					
+					if(entity != null && entity instanceof LivingEntity && entity.getPassenger() != player 
+							&& entity != player && !(entity instanceof Wither) && !(entity instanceof EnderDragon)) {
+						List<WrappedWatchableObject> list = event.getPacket().getWatchableCollectionModifier().read(0);
+						for(WrappedWatchableObject wwo : list) {
+							if(wwo.getIndex() == 7 && ((Float)wwo.getValue()) > 0) {
+								wwo.setValue(20f);
 							}
-				    	}
+						}
+						event.getPacket().getWatchableCollectionModifier().write(0, list);
 					}
-	  	        }
-				catch (Exception e){
-	  	          e.printStackTrace();
-	  	        }
+				}
+			});
+	}
+	
+	private Entity getEntityById(World world, int id) {
+		for(Entity e : world.getEntities()) {
+			if(e.getEntityId() == id) {
+				return e;
 			}
-		});
+		}
+		return null;
 	}
-    
-    private boolean shouldBeObfuscated(Material type) {
```

Comment:
The functionality of this should stay, but making it a hashset would be a lot better